learn-ocaml.el

```diff
@@ -197,4 +269,71 @@
      (learn-ocaml-give-server
       (lambda (server)
 	(learn-ocaml-on-load-to-wrap token server))))))
-  
+
+
+
+;;;;;;;;; menu definition ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+(defvar learn-ocaml-mode-map
+  (let ((map (make-sparse-keymap)))
+    (define-key map "\C-as" #'learn-ocaml-show-metadata)
+    (define-key map [menu-bar] nil )
+    map))
+
+(easy-menu-define learn-ocaml-mode-menu
+  learn-ocaml-mode-map
+  "Learnocaml Mode Menu."
+  '("Learnocaml"
+    ["Show metadata" learn-ocaml-show-metadata]
+    ["Change server" learn-ocaml-change-server]
+    ["Change token" learn-ocaml-change-token]
+    ["Create token" learn-ocaml-create-token-wrapper]
+    ["Grade" learn-ocaml-grade-wrapper]
+    ["Download server version" learn-ocaml-download-server-file-wrapper]
+    ))
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+
+;;;;id management ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+(defun learn-ocaml-update-exercise-id-view ()
+  (define-key-after
+    learn-ocaml-mode-map
+    [menu-bar exercise-id]
+     `(,(concat "exercise-id: " learn-ocaml-exercise-id) .
+	  ,(make-sparse-keymap "Exercise-id"))
+     )
+  (define-key
+    learn-ocaml-mode-map
+    [menu-bar exercise-id reset]
+    '("Reset id" . learn-ocaml-exercise-id-initializer))
+  (define-key 
+    learn-ocaml-mode-map
+    [menu-bar exercise-id change]
+    '("Change id" . learn-ocaml-change-exercise-id ))
+  (force-mode-line-update)) ;; a random instruction is needed to update menu bar
+
+(defun learn-ocaml-exercise-id-initializer()
+  (interactive)
+  (setq-local learn-ocaml-exercise-id
+	(file-name-sans-extension(file-name-base  buffer-file-name)))
+  (learn-ocaml-update-exercise-id-view))
+
+(defun learn-ocaml-change-exercise-id (new-id)
+  (interactive "sEnter new id : ")
+  (setq-local learn-ocaml-exercise-id new-id)
+  (learn-ocaml-update-exercise-id-view))
+
+(add-hook 'window-configuration-change-hook #'learn-ocaml-update-exercise-id-view)
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+
+;;;;;;;;;;;;;;;;;;;;;definition of the mode ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+
+(define-minor-mode learn-ocaml-mode
```

Comment:
A standard practice consists in adding the line

```
;;;###autoload
```

here, just before the line ` (define-minor-mode learn-ocaml-mode`, so the interactive function `learn-ocaml-mode` will be exposed without needing to require the `learn-ocaml` file itself; the user can just do `M-x learn-ocaml-mode RET` to load the whole `.el` file.
cf. https://www.gnu.org/software/emacs/manual/html_node/elisp/Autoload.html